### ISSUE TITLE
calc.py throws erros after building SasView

### ISSUE THREAD

**Issue by sl-saxs** [SasView/sasview#1583]
I've just tried running SasView, downloaded from the ESS_GUI branch. 
OS is LinuxMint 18. 
running "sudo python3 setup.py" works fine. 
Upon running "sudo python3 run.py", I get an error message from
./src/sas/sascalc/pr/calc.py (see below). 
I modified "calc.py" like this:
- modifications on the following lines: 209, 229, 248, 269, 296, 320, 349
- in all functions on these lines, the @njit input variable type was changed from "u8" to "i8"
- funny enough, there was no errors for the other functions using the @njit input variable type "u8" (why?)
Running "sudo python3 setup.py" and "sudo python3 run.py" again starts the software just fine.

And here's the error message:
'''
Traceback (most recent call last):
  File "run.py", line 138, in <module>
    run_sasview()
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/qtgui/MainWindow/MainWindow.py", line 102, in run_sasview
    mainwindow = MainSasViewWindow(screen_resolution)
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/qtgui/MainWindow/MainWindow.py", line 34, in __init__
    from .GuiManager import GuiManager
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/qtgui/MainWindow/GuiManager.py", line 50, in <module>
    import sas.qtgui.Perspectives as Perspectives
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/qtgui/Perspectives/__init__.py", line 6, in <module>
    from .Inversion.InversionPerspective import InversionWindow
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/qtgui/Perspectives/Inversion/InversionPerspective.py", line 15, in <module>
    from sas.sascalc.pr.invertor import Invertor
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/sascalc/pr/invertor.py", line 29, in <module>
    from .p_invertor import Pinvertor
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/sascalc/pr/p_invertor.py", line 12, in <module>
    from . import calc
  File "/home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/sascalc/pr/calc.py", line 209, in <module>
    @njit('f8(f8[:], f8, u8)')
  File "/usr/local/lib/python3.6/dist-packages/numba/core/decorators.py", line 198, in wrapper
    disp.compile(sig)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/dispatcher.py", line 794, in compile
    cres = self._compiler.compile(args, return_type)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/dispatcher.py", line 81, in compile
    raise retval
  File "/usr/local/lib/python3.6/dist-packages/numba/core/dispatcher.py", line 91, in _compile_cached
    retval = self._compile_core(args, return_type)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/dispatcher.py", line 109, in _compile_core
    pipeline_class=self.pipeline_class)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler.py", line 568, in compile_extra
    return pipeline.compile_extra(func)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler.py", line 339, in compile_extra
    return self._compile_bytecode()
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler.py", line 401, in _compile_bytecode
    return self._compile_core()
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler.py", line 381, in _compile_core
    raise e
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler.py", line 372, in _compile_core
    pm.run(self.state)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_machinery.py", line 341, in run
    raise patched_exception
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_machinery.py", line 332, in run
    self._runPass(idx, pass_inst, state)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_lock.py", line 32, in _acquire_compile_lock
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_machinery.py", line 291, in _runPass
    mutated |= check(pss.run_pass, internal_state)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/compiler_machinery.py", line 264, in check
    mangled = func(compiler_state)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/typed_passes.py", line 98, in run_pass
    raise_errors=self._raise_errors)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/typed_passes.py", line 70, in type_inference_stage
    infer.propagate(raise_errors=raise_errors)
  File "/usr/local/lib/python3.6/dist-packages/numba/core/typeinfer.py", line 986, in propagate
    raise errors[0]
numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(<built-in function arange>) with argument(s) of type(s): (uint64)
 * parameterized
In definition 0:
    TypeError: '>' not supported between instances of 'Integer' and 'Integer'
    raised from /usr/local/lib/python3.6/dist-packages/numba/np/arrayobj.py:3837
In definition 1:
    TypeError: '>' not supported between instances of 'Integer' and 'Integer'
    raised from /usr/local/lib/python3.6/dist-packages/numba/np/arrayobj.py:3837
This error is usually caused by passing an argument of a type that is unsupported by the named function.
[1] During: resolving callee type: Function(<built-in function arange>)
[2] During: typing of call at /home/sebastian/GITHUB/SasView_ESS_GUI/sasview-ESS_GUI/sasview-ESS_GUI/src/sas/sascalc/pr/calc.py (221)

File "src/sas/sascalc/pr/calc.py", line 221:
def reg_term(pars, d_max, nslice):
    <source elided>
    nslice_d = np.float64(nslice)
    i_r = np.arange(nslice)

'''

**Comment by andyfaff** [comment]
Is the `sudo` required here?

**Comment by sl-saxs** [comment]
> Is the `sudo` required here?
No. 'sudo' is not necessary here.
So, running "python3 setup.py build" and "python3 run.py" is enough.
But that has no influence on the errors that are being thrown from "src/sas/sascalc/pr/calc.py".
However, if SasView should be installed in 'usr/local/lib/"pythonX.Y"/dist-packages/'
one needs to run "sudo python3 setup.py install"
SasView can also be installed in "/path/you/want/for/installation/" with:
"python3 setup.py install --home=/path/you/want/for/installation/"

**Comment by pkienzle** [comment]
Reproduced with miniconda using numba 0.49.1 and llvmlite 0.32.1 on Ubuntu 18.04:
```python
.../sasview$ PYTHONPATH=src python
>>> import numba
>>> print(numba.__version__)
0.49.1
>>> import sas.sascalc.pr.calc
...
```
This is directly from the repo, with no u8/i8 changes.

**Comment by sl-saxs** [comment]
I've used:
numba, version 0.50.0
llvmlite, version 0.31.0

**Comment by pkienzle** [comment]
Please try PR #1585. I didn't fiddle with the int types since they weren't causing me problems.

**Comment by butlerpd** [comment]
@pkienzle This looks like it was fixed/merged for 5.0.3. Can we close the issue?
